### ISSUE TITLE
Issue converting to `%`

### ISSUE THREAD

**Issue by lafrech** [hgrecco/pint#1758]
I'm trying to use the new `%` unit introduced in #1661.

While it works to define a quantity, that can then be converted to another dimensionless unit:

```
>>> ureg.Quantity(1, "%")
<Quantity(1, 'percent')>
>>> ureg.Quantity(1, "%").m_as("")
0.01
```

it doesn't seem to work the other way around.

```
>>> ureg.Quantity(1, "")
<Quantity(1, 'dimensionless')>
>>> ureg.Quantity(1, "").m_as("%")
pint.errors.DefinitionSyntaxError: missing unary operator "%"
```

Is this a known limitation? It doesn't seem consistent.

Looks like preprocessors apply to quantity definitions but not unit expressions in `Quantity.to`.

I think I have a workaround that at least solves my use case:

```
>>> ureg.Quantity(1, "").m_as(ureg.Unit("%"))
100.0
```

Perhaps this cast could be done somewhere in `m_as`, `to` or further down the way.

**Comment by hgrecco** [comment]
Have you pull the latest version?

**Comment by hgrecco** [comment]
confirmed

**Comment by mmarras** [comment]
I think this can be closed with 0.23, no? I tried `ureg.Quantity(1,'').to('%')` and it works now.

**Comment by mmarras** [comment]
> confirmed

I think this can be closed with 0.23, no? I tried ureg.Quantity(1,'').to('%') and it works now.

**Comment by lafrech** [comment]
Indeed, this seems to be fixed in 0.23.0.

Thanks.
